src/app/components/home/home.component.ts

```diff
@@ -8,4 +18,53 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class HomeComponent {
+  private featuredCategoriesIds: string[] = [CategoryId.fruitsAndVegetables, CategoryId.snackAndMunchies];
+
+  readonly model$: Observable<HomeQueryModel> = combineLatest([
+    this._categoryService.getAll(),
+    this._storeService.getAllWithTags(),
+    this._productService.getAll()
+  ]).pipe(
+    map(
+      ([categories, stores, products]: [CategoryModel[], StoreModel[], ProductModel[]]) => this._mapQueryModel(categories, stores, products)
+    )
+  );
+
+  constructor(private _categoryService: CategoryService, private _storeService: StoreService, private _productService: ProductService) {
+  }
+
+  private _mapQueryModel(categories: CategoryModel[], stores: StoreModel[], products: ProductModel[]): HomeQueryModel {
```

Comment:
this function is pretty long, how about we split it into smaller private functions?